packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_refactors.dart

```diff
@@ -0,0 +1,228 @@
+                    setState(() {
+                      _showAllRefactors = !_showAllRefactors;
+                    });
+                  },
+                ),
+              ),
+            ],
+          ),
+        const PaddedDivider.noPadding(),
+      ],
+    );
+  }
+
+  void _categorizeRefactors() {
+    _mainRefactorsGroup.clear();
+    _otherRefactorsGroup.clear();
+    for (final refactor in widget.refactors) {
+      // Ignore any unexpected refactors that don't begin with "Wrap with".
+      if (!refactor.title.startsWith('Wrap with')) continue;
+
+      final category = _mainRefactors.contains(refactor.title)
+          ? _mainRefactorsGroup
+          : _otherRefactorsGroup;
+      category.add(refactor);
+    }
+  }
+}
+
+/// A button which triggers a single "Wrap with" refactor.
+///
+/// Buttons for refactors in [_refactorsWithIconAsset] have an icon.
+class _WrapWithButton extends StatelessWidget {
+  const _WrapWithButton(
+    this.action, {
+    required this.executeCommand,
+    required this.iconOnly,
+  });
+
+  final CodeActionCommand action;
+  final ExecuteCommandFunction executeCommand;
+  final bool iconOnly;
+
+  static const _iconAssetPath = 'icons/property_editor/';
+
+  String get label {
+    final wrapperName = action.title.split('Wrap with ').last;
+    return wrapperName == 'widget...' ? 'Widget' : wrapperName;
+  }
+
+  String get command => action.command;
+
+  String? _iconAsset({bool darkMode = false}) {
+    if (!_refactorsWithIconAsset.contains(action.title)) {
+      return null;
+    }
+
+    final path = '$_iconAssetPath${label.toLowerCase()}';
+    return '${path}_${darkMode ? 'dark' : 'light'}_theme.png';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final iconAsset = _iconAsset(darkMode: theme.isDarkTheme);
+
+    final button = OutlinedButton(
+      style: OutlinedButton.styleFrom(
+        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(4.0)),
+        side: BorderSide(
+          color: iconOnly ? Colors.transparent : theme.colorScheme.onSurface,
+        ),
+        padding: const EdgeInsets.all(densePadding),
+        tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+      ),
+      child: iconAsset != null
+          ? Row(
+              mainAxisSize: MainAxisSize.min,
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                Image.asset(iconAsset, height: actionsIconSize),
+              ],
+            )
+          : _buttonText(label, theme: theme),
+      onPressed: () async {
+        await executeCommand(action);
+      },
+    );
+
+    return Padding(
+      padding: EdgeInsets.all(iconOnly ? densePadding / 2 : densePadding),
+      child: iconOnly ? DevToolsTooltip(message: label, child: button) : button,
+    );
+  }
+
+  Text _buttonText(String label, {required ThemeData theme}) {
+    // Show the first letter of the label as the icon.
```

Comment:
should we assert that label.isNotEmpty here?

Reply:
Done!